packages/eslint-plugin-query/src/rules/infinite-query-property-order/constants.ts

```diff
@@ -0,0 +1,13 @@
+export const infiniteQueryFunctions = [
+  'infiniteQueryOptions',
+  'useInfiniteQuery',
+  'useSuspenseInfiniteQuery',
+] as const
+
+export type InfiniteQueryFunctions = (typeof infiniteQueryFunctions)[number]
+
+export const checkedProperties = [
+  'queryFn',
+  'getPreviousPageParam',
+  'getNextPageParam',
+] as const
```

Comment:
is this the full order that we must have in the object? Because it shouldn't matter if `getNextPageParam` comes before or after `getPreviousPageParam` - it only matters that they are after `queryFn`. So maybe it needs like a dependency tree:

```
const propertyDependencies = {
  queryFn: ['getPreviousPageParam', 'getNextPageParam']
}
```

this would define that `'getPreviousPageParam'` and `'getNextPageParam'` depend on `queryFn` and thus must be after it. What do you think?

Reply:
yes ideally a tree like structure would be best. I decided against this in router (similar issue with validateSearch and params.parse) to keep this plugin simple.

Reply:
ok fine by me for now